app/services/catalog/update_order_item.rb

```diff
@@ -0,0 +1,56 @@
+module Catalog
+  class UpdateOrderItem
+    class ServiceInstanceWithoutExternalUrl < StandardError; end
+
+    def initialize(topic)
+      @payload = topic.payload
+      @message = topic.message
+    end
+
+    def process
+      Rails.logger.info("Processing service order topic message: #{@message} with payload: #{@payload}")
+
+      Rails.logger.info("Searching for OrderItem with a task_ref: #{@payload["task_id"]}")
+      order_item = find_order_item
+      Rails.logger.info("Found OrderItem: #{order_item.id}")
+
+      ManageIQ::API::Common::Request.with_request(order_item.context.transform_keys(&:to_sym)) do
+        order_item.update_message("info", "Task update message received with payload: #{@payload}")
```

Comment:
@eclarizio Should this be in a separate function, manage_state or something like that